python/ray/experimental/util/types.py

```diff
@@ -9,11 +9,28 @@ class _CollectiveOp(Enum):
 
 @PublicAPI
 class ReduceOp(_CollectiveOp):
+    pass
+
+
+@PublicAPI
+class AllReduceOp(ReduceOp):
+    SUM = 0
+    PRODUCT = 1
+    MAX = 2
+    MIN = 3
+    AVG = 4
```

Comment:
Define these on ReduceOp instead. We want AllReduceOp and ReduceScatterOp to share the same binary operations.